File: diploma.client/src/model/PersonDisplay.ts
import {UserCan} from "./UserCan";

export class PersonDisplay {
  public fio: string;
  public username: string;
  public role: string;
  public cans: UserCan[];

  public static of(a: any): PersonDisplay {
    const ret = new PersonDisplay();
    ret.assign(a);
    return ret;
  }

  assign(a: any) {
    this.fio = a.fio;
    this.username = a.username;
    this.role = a.role;
    // noinspection SuspiciousInstanceOfGuard
    this.cans = (a.cans instanceof Array) ? a.cans.map(c => c) : [];
  }
}

File: diploma.controller/src/kz/greetgo/diploma/controller/model/PersonDisplay.java
package kz.greetgo.diploma.controller.model;

import java.util.ArrayList;
import java.util.List;

public class PersonDisplay {

	public String fio;

	public String username;

	public String role;

	public List<UserCan> cans = new ArrayList<>();

	@Override
	public String toString() {

		return "PersonDisplay{" +
			"fio='" + fio + '\'' +
			", username='" + username + '\'' +
			", role='" + role + '\'' +
			", cans=" + cans +
			'}';
	}
//The following code would be not removed after regenerating
	///LEAVE_FURTHER
}
